src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -1858,12 +1858,12 @@ class WhenClauseSegment(BaseSegment):
             Ref("ExpressionSegment"),
             Dedent,
         ),
-        Indent,
+        Conditional(Indent, indented_then=True),
         "THEN",
         ImplicitIndent,
         Ref("ExpressionSegment"),
         Dedent,
-        Dedent,
+        Conditional(Dedent, indented_then=True),
```

Comment:
This is a super nitpick, but shouldn't the Dedent above this be the conditional one?

Reply:
@fredriv - would you mind shifting this as per the suggestion here? We've probably got a day or two before the other PR is reviewed and merged so whenever you get a moment.

Reply:
Hmm, having a closer look - the `Dedent` on line 1865 matches the `ImplicitIndent` on line 1863, so I think it's correct the way it is? 🤔 What do you think @WittierDinosaur @alanmcruickshank?